### PR TITLE
Poulomi/login signup activity

### LINKED ISSUE #16: Create Login/Signup Activity

### LINKED ISSUE #23: Add Login + Signup functionality

### PR DESCRIPTION
**Pull request by poulomi-g**
Everything implemented except support for adding user info to backend. All intent tests passing.

### CODE REVIEW

**Comment by 99patricia** [comment]
@poulomi-g by adding the user info to the backend, is this the signup with username part and adding it so that users can log in using username?

**Comment by poulomi-g** [comment]
@99patricia At the moment, only email login is allowed since backend database + functionality isnt set up. But signup allows username and email entry both. Once the users collection is created I can work on that functionality

**Comment by 99patricia** [comment]
> @poulomi-g by adding the user info to the backend, is this the signup with username part and adding it so that users can log in using username?

[this](https://stackoverflow.com/a/46108298) might be a good approach for this, but we can worry about that in another PR

**Comment by poulomi-g** [comment]
> > @poulomi-g by adding the user info to the backend, is this the signup with username part and adding it so that users can log in using username?
> 
> [this](https://stackoverflow.com/a/46108298) might be a good approach for this, but we can worry about that in another PR

I thought of this too! I'll implement this as soon as backend is ready

**Comment by 99patricia** [comment]
@poulomi-g this is looking great! I just have one comment, I'm thinking all our activities should be named `____Activity.java` just to be consistent so maybe change the `Signup.java` activity to `SignupActivity.java` and same thing with `LoginConfirmation`

**Comment by poulomi-g** [comment]
@99patricia Makes sense! Latest commit takes care of it. Dont merge yet, @amahesh19 and I will try to figure out the backend functionality rn

File: app/build.gradle [hunk 1]
```diff
@@ -48,4 +48,8 @@ dependencies {
     testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.0.1'
     androidTestImplementation 'com.jayway.android.robotium:robotium-solo:5.3.1'
```
Comment (99patricia) [inline app/build.gradle:53]: idk if it's just on my machine, but without these lines in the build.gradle
```
    testImplementation 'junit:junit:4.12'
    androidTestImplementation 'androidx.test.ext:junit:1.1.1'
    androidTestImplementation 'androidx.test:runner:1.2.0'
    androidTestImplementation 'androidx.test:rules:1.2.0'
    androidTestImplementation 'androidx.test.espresso:espresso-core:3.2.0'
```
the tests complain about import errors

Comment (99patricia) [reply app/build.gradle:53]: before the UI tests we also have to ensure the user is logged out/on the login screen bc the tests are failing for me when I'm logged in

**Pull request merged**